Mathlib/CategoryTheory/Monoidal/Transport.lean

```diff
@@ -35,117 +40,193 @@ variable {C : Type u₁} [Category.{v₁} C] [MonoidalCategory.{v₁} C]
 
 variable {D : Type u₂} [Category.{v₂} D]
 
--- porting note: it was @[simps {attrs := [`_refl_lemma]}]
-/-- Transport a monoidal structure along an equivalence of (plain) categories.
+/--
+Induce the lawfulness of the monoidal structure along an faithful functor of (plain) categories,
+where the operations are already defined on the destination type `D`.
+
+The functor `F` must preserve all the data parts of the monoidal structure between the two
+categories.
+
+Note that `μIsoSymm` and `εIsoSymm` correspond to the reversed versions of
+`CategoryTheory.LaxMonoidalFunctor.μIso` and `CategoryTheory.LaxMonoidalFunctor.εIso`.
 -/
-@[simps]
-def transport (e : C ≌ D) : MonoidalCategory.{v₂} D where
-  tensorObj X Y := e.functor.obj (e.inverse.obj X ⊗ e.inverse.obj Y)
-  whiskerLeft := fun X _ _ f ↦ e.functor.map (e.inverse.obj X ◁ e.inverse.map f)
-  whiskerRight := fun f X ↦ e.functor.map (e.inverse.map f ▷ e.inverse.obj X)
-  tensorHom_def := by simp [tensorHom_def]
-  tensorHom f g := e.functor.map (e.inverse.map f ⊗ e.inverse.map g)
-  tensorUnit' := e.functor.obj (𝟙_ C)
-  associator X Y Z :=
-    e.functor.mapIso
-      (((e.unitIso.app _).symm ⊗ Iso.refl _) ≪≫
-        α_ (e.inverse.obj X) (e.inverse.obj Y) (e.inverse.obj Z) ≪≫ (Iso.refl _ ⊗ e.unitIso.app _))
-  leftUnitor X :=
-    e.functor.mapIso (((e.unitIso.app _).symm ⊗ Iso.refl _) ≪≫ λ_ (e.inverse.obj X)) ≪≫
-      e.counitIso.app _
-  rightUnitor X :=
-    e.functor.mapIso ((Iso.refl _ ⊗ (e.unitIso.app _).symm) ≪≫ ρ_ (e.inverse.obj X)) ≪≫
-      e.counitIso.app _
-  triangle X Y := by
+@[simps
+  tensorObj whiskerLeft whiskerRight tensorHom tensorUnit' associator leftUnitor rightUnitor]
+abbrev induced (F : D ⥤ C) [Faithful F]
```

Comment:
I would think the code would look nicer if the parameters of `induced` where part of a structure named something like `CoreInducedMonoidalCategory`. Otherwise, it is great! (This construction shall certainly have other applications in the future.)

Reply:
I tried this in 69af7ff198b82106d76fb281119eb5a6b4aab4c1, but it seems to cause reducibility problems that break `aesop_cat` and force some nasty `erw`s.

I have some other ideas for cleaning this up (such as moving the operators of `MonoidalCategory` into a data-only typeclass), but they're bigger refactors that I'd rather not make at the same time as this.

Are you happy with the structure-free approach for now?

Reply:
The need of `erw` is a quite common general issue with full subcategories, because `Hom` types are defeq accross different categories. I would very much prioritize the use of a structure, even if there are a few proofs which require a trivial fix.

Reply:
I was able to get aesop to work by adding two local simp lemmas.

Reply:
I expect the structure version to cause quite a large performance regression in `RepresentationTheory`; I'd prefer to merge the version with the individual fields, and leave the structure refactor until after the `MonoidalCategory` refactor in #6307 concludes, as otherwise I worry that I'm just going to trap that refactor in timeouts.

I've started another benchmark to see just how much worse things are...